lib/models/package-info-cache/package-info.js

```diff
@@ -415,6 +421,190 @@ class PackageInfo {
 
     return (this.addonConstructor = ctor);
   }
+
+  /**
+   * Construct an addon instance.
+   * NOTE: this does NOT call constructors for the child addons. That is left to
+   * the caller to do, so they can insert any other logic they want.
+   * @private
+   * @method constructAddonInstance
+   * @param {Project|Addon} parent the parent that directly contains this addon
+   * @param {Project} project the project that is/contains this addon
+   */
+  constructAddonInstance(parent, project) {
+    let start = Date.now();
+
+    let AddonConstructor = this.getAddonConstructor();
+
+    let addon;
+
+    try {
+      addon = new AddonConstructor(parent, project);
+    } catch (e) {
+      if (parent && parent.ui) {
+        parent.ui.writeError(e);
+      }
+      const SilentError = require('silent-error');
+      throw new SilentError(`An error occurred in the constructor for ${this.name} at ${this.realPath}`);
+    }
+
+    AddonConstructor._meta_.initializeIn = Date.now() - start;
+    addon.constructor = AddonConstructor;
+
+    return addon;
+  }
+
+  /**
+   * Given a parent object of a potential addon, go up the parent chain to
+   * find the addon's bundle root (an object that returns 'isBundleRoot = true'.)
+   * In general that is the Project or a lazy engine.
+   * Projects are always bundle roots, so this should always pass. Just in case,
+   * if it does not, throw an error. "Lazy" engines will also be bundle
+   * roots.
+   *
+   * @method findBundleRoot
+   * @param {Project|Addon} parent
+   * @return {Object} the object in the 'parent' chain that is a bundle root.
+   */
+  findBundleRoot(parent) {
+    let tmp = parent;
+    while (tmp) {
+      if (tmp.isBundleRoot) {
```

Comment:
Let's add a small utility that checks lazyLoading (not `isBundleRoot`) here.

Something like:

```js
if (isBundleRoot(tmp)) {
```